src/api/app/models/kiwi/image.rb

```diff
@@ -104,16 +104,17 @@ def to_xml
 
     return nil unless image && image.first_element_child
 
-    doc.xpath("image/packages").remove
-    xml_packages = package_groups.map(&:to_xml).join("\n")
-    image.first_element_child.after(xml_packages)
+    # for now we only write the default package group
+    xml_packages = default_package_group.to_xml
+    packages = doc.xpath('image/packages[@type="image"]').first
+    packages ? packages.replace(xml_packages) : image.last_element_child.after(xml_packages)
```

Comment:
Why not an ```if ... else ... end``` ?? This is a little bit hard to understand because is not used to assign the value to anything and this is running methods instead.

Reply:
Is this explicitly tested?

Reply:
Done!
Yes, I use one time the DEFAULT_KIWI_BODY (without packages and repositories) and the one time the kiwi_xml provided from ``include_context 'a kiwi image xml'`` which already contains packages and repositories!